umpost/conversion_driver_esm1p5.py

```diff
@@ -0,0 +1,201 @@
+#!/usr/bin/env python3
+"""
+ESM1.5 conversion driver
+
+Wrapper script for automated fields file to NetCDF conversion
+during ESM1.5 simulations. Runs conversion module (currently 
+um2netcdf4) on each atmospheric output in a specified directory.
+
+Adapted from Martin Dix's conversion driver for CM2: 
+https://github.com/ACCESS-NRI/access-cm2-drivers/blob/main/src/run_um2netcdf.py
+"""
+
+
+import os
+import collections
+import um2netcdf4
+import re
+import f90nml
+import warnings
+import argparse
+import errno
+from pathlib import Path
+
+
+# TODO: um2netcdf will update the way arguments are fed to `process`.
+# https://github.com/ACCESS-NRI/um2nc-standalone/issues/17
+# Update the below arguments once the changes are added.
+
+# Named tuple to hold the argument list
+ARG_NAMES = collections.namedtuple(
+    "Args",
+    "nckind compression simple nomask hcrit verbose include_list exclude_list nohist use64bit",
+)
+# TODO: Confirm with Martin the below arguments are appropriate defaults.
+ARG_VALS = ARG_NAMES(3, 4, True, False, 0.5, False, None, None, False, False)
+
+# TODO: um2nc standalone will raise more specific exceptions.
+# See https://github.com/ACCESS-NRI/um2nc-standalone/issues/18
+# Improve exception handling here once those changes have been made.
+ALLOWED_UM2NC_EXCEPTION_MESSAGES = {
+    "TIMESERIES_ERROR": "Variable can not be processed",
+}
+
+
+def set_esm1p5_fields_file_pattern(run_id):
+    """
+    Generate regex pattern for finding current experiment's UM outputs.
+
+    Parameters
+    ----------
+    run_id : 5 character run ID for the current UM simulation.
```

Comment:
Is it necessary to ensure the `run_id` is a 5 char string in this func?

Reply:
Good point. `run_id` produced by the version of the UM used by ESM1.5 should be a 5 character string. I've added in a value error in case something goes wrong here.